ports/esp32/machine_touchpad.c

```diff
@@ -56,6 +57,73 @@ STATIC const mtp_obj_t touchpad_obj[] = {
     {{&machine_touchpad_type}, GPIO_NUM_33, TOUCH_PAD_NUM8},
     {{&machine_touchpad_type}, GPIO_NUM_32, TOUCH_PAD_NUM9},
 };
+
+STATIC mp_obj_t mtp_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw,
```

Comment:
It looks like this function and the functions below are duplicated/copied directly from the existing ones.  Why is it done that way?  It would be much better to just have one version of these functions and use `#if`'s to configure it for S2/S3/etc.

Reply:
Yes, thanks for the review.
I tidied up the code and everything became clearer.
Updated and force-pushed.

Reply:
Great, that's looking a lot better now!